### PR TITLE
fix: Space.Compact when wrapping a single child component

### PR DESCRIPTION
**Pull request by foryuki**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/38590

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Space.Compact when wrapping a single child component.       |
| 🇨🇳 Chinese |     Space.Compact 包裹单个子组件时，展示该子组件本身的样式，不进行紧凑模式的样式定制。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
Test failed. Please check.

File: components/style/compact-item-vertical.tsx [hunk 1]
```diff
@@ -27,14 +27,14 @@ function compactItemBorderVerticalRadius(prefixCls: string): CSSObject {
       borderRadius: 0,
     },
 
-    '&-item&-first-item': {
+    '&-item&-first-item:not(&-last-item)': {
```
Comment (afc163) [inline components/style/compact-item-vertical.tsx:30]: 能用 `:only-child` 代替么

Comment (foryuki) [reply components/style/compact-item-vertical.tsx:30]: 可以用 :only-child 代替的，看起来理解成本没啥变化？
image

Comment (afc163) [reply components/style/compact-item-vertical.tsx:30]: 这样是不是不需要加 ant-btn-compact-first-item ant-btn-compact-last-item 这些 className 

直接用 :first-child  :last-child  :only-child

Comment (foryuki) [reply components/style/compact-item-vertical.tsx:30]: 如果是直接设置唯一子元素的边框的话，那用 :only-child 会简洁很多。

此处的逻辑是 「重置」  border-radius 为 0 ，所以得要这一大串呢。

**Review by afc163 — APPROVED** [review]

**Pull request merged**